src/main/scala/workflow/GatherTransformer.scala

```diff
@@ -0,0 +1,15 @@
+package workflow
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+private[workflow] class GatherTransformer[T] extends TransformerNode[Seq[T]] {
+  def transform(dataDependencies: Seq[_], fitDependencies: Seq[TransformerNode[_]]): Seq[T] = dataDependencies.map(_.asInstanceOf[T])
+
+  def transformRDD(dataDependencies: Seq[RDD[_]], fitDependencies: Seq[TransformerNode[_]]): RDD[Seq[T]] = {
+    dataDependencies.map(_.asInstanceOf[RDD[T]].map(t => Seq(t))).reduceLeft((x, y) => {
+      x.zip(y).map(z => z._1 ++ z._2)
```

Comment:
This is probably not a big deal, but from what I can see this sets up a lineage of length `n` if we have `n` dataDependencies. I think we can make it `log n` if we used a tree-like structure, but it might not matter for our use cases